aten/src/ATen/native/quantized/cpu/qnnpack/src/average-pooling.c

```diff
@@ -270,6 +270,67 @@ enum pytorch_qnnp_status pytorch_qnnp_setup_average_pooling2d_nhwc_q8(
    * buffer */
   const uint32_t mr = pytorch_qnnp_params.q8avgpool.mr;
 
+  /*
+   * Indirection buffer:
+   * Imagine a we want to do dw conv or avgpooling with these parameters:
+   * kernel_width/height=3 stride=2
+   * Input is:
+   *  ---------------
+   *  |0|1|2|3|4|5|6|
+   *  ---------------       -------
+   *  | | | | | | | |   to  |0|1|2|
+   *  ---------------       -------
+   *  | | | | | | | |       | | | |
+   *  ---------------       -------
+   *  | | | | | | | |
+   *  ---------------
+   *  | | | | | | | |
+   *  ---------------
+   *
+   *  Thus we are going from width=7 height=5 input to height=2 width=3
```

Comment:
Why did you switch from width,height to height,width?  Is that an error?

Reply:
Thanks for the catch. I will make it consistent.